src/matchers.rs

```diff
@@ -7,24 +7,27 @@
 use serde_json::Value;
 use std::iter;
 
+/// An iterator over matcher selection results.
```

Comment:
We should probably add rustdoc for all public items, in a separate PR. wdyt?

Reply:
yep I noticed as well, but refrained from doing the change in the same PR

Reply:
ok